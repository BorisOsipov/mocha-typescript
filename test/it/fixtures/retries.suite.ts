import { assert } from "chai";
import { only, pending, retries, skip, slow, suite, test, timeout } from "../../../index";

@suite class FlakyMethodDecorator {
    private static runCount1 = 0;
    private static runCount2 = 0;
    private static runCount3 = 0;

    @test @retries(1) public tryOnce() {
        FlakyMethodDecorator.runCount1++;
        assert.isAbove(FlakyMethodDecorator.runCount1, 2);
    }

    @test @retries(2) public tryTwice() {
        FlakyMethodDecorator.runCount2++;
        assert.isAbove(FlakyMethodDecorator.runCount2, 2);
    }

    @test @retries(3) public tryTrice() {
        FlakyMethodDecorator.runCount3++;
        assert.isAbove(FlakyMethodDecorator.runCount3, 2);
    }
}

@suite @retries(3) class FlakySuiteDecorator {
    private static runCount1 = 0;
    private static runCount2 = 0;
    private static runCount3 = 0;

    @test public tryToGetPass2() {
        FlakySuiteDecorator.runCount1++;
        assert.isAbove(FlakySuiteDecorator.runCount1, 2);
    }

    @test public tryToGetPass4() {
        FlakySuiteDecorator.runCount2++;
        assert.isAbove(FlakySuiteDecorator.runCount2, 4);
    }

    @test @retries(4) public overrideSuiteRetries() {
        FlakySuiteDecorator.runCount3++;
        assert.isAbove(FlakySuiteDecorator.runCount3, 4);
    }
}
