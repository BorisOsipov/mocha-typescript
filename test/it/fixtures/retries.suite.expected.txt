
  FlakyMethodDecorator
    1) tryOnce
    ✓ tryTwice
    ✓ tryTrice

  FlakySuiteDecorator
    ✓ tryToGetPass2
    2) tryToGetPass4
    ✓ overrideSuiteRetries

  4 passing
  2 failing

  1) FlakyMethodDecorator
       tryOnce:

      AssertionError: expected 2 to be above 2

  2) FlakySuiteDecorator
       tryToGetPass4:

      AssertionError: expected 4 to be above 4
